### DISCUSSION TITLE
Idea: Folderstructure to Albumstructure

### DISCUSSION THREAD

**Discussion by ISeeTWizard** [photoprism/photoprism#2269] [Ideas]
Hi
I searched a bit and didn't found that question till now (maybe I oversee something)?

So what would be great if as example a folder structure could be imported also directly as albums and so have album x and under it different other albums.
So as example I have a structure per year and than per day automatically generated with EXIF sorter.
I can create manually the albums from the folders but with several thousand folders I get nuts...
As, if I see correctly, calendar etc depends on albums...

Also a multiple user login would be great (just a login - not other personal photos - so not another instance like I already read here)

For the rest I'm waiting that my photos are all imported and face recognition etc has started (takes a while on a Synology NAS :) )

**Comment by graciousgrey** [comment]
Why do you want to convert your folders into albums? 

You can browse and share your folders (https://docs.photoprism.app/user-guide/organize/folders/) just like you can browse albums (https://docs.photoprism.app/user-guide/organize/albums/). The difference is that albums are virtual and user generated, whereas folders display the state of your filesystem. 

Note that it is not possible to have sub-albums whereas a hierarchical view for the folder section is planned.

The calendar view consists of autogenerated albums based on the date information from your photos metadata.

Multi user functionalities are in progress. Adding multiple users with different roles to one library will be part of it.

**Reply by ISeeTWizard** [reply]
Hi
ok - as in calendar I had no photos and it was written there are no albums for me it was so that I had to create albums
maybe there is nothing because the scan is still running

thanks for your answer already - I wait till the end of the scan and see how it looks than.

**Reply by graciousgrey** [reply]
That's expected calendar, states and moments are created after the scan is done :)

**Reply by ISeeTWizard** [reply]
Thanks - I missed that point :( Boah and that for an IT technician 
I read about face recognition etc but not for the calendar...
Sorry about that :/

I wait till the end and continue my tests than :)

Until now it looks good - not 100% what I want but at maybe 98% 
Tested so many software and hardware combinations and most are not good or not maintained anymore - even those I once payed for - But over the years I think normal...

I love this one already as it's running in a docker and doesn't move my photos around to another folder structure etc.

**Comment by ISeeTWizard** [comment]
Hi so far I'm happy and so I'm started with a sponsoring
All is imported etc. :)

I swapped to German but the categories were created in English - I simply can rename them? I don't think that this has any side effects?
Also some are simply wrong (an industrial building is as example a weapon lol) should I rename those or simply delete and recreate a new one?

The only thing I definitely miss are sub albums or so...
Even under folders it's not displayed like the folders are but just on one level...

Same would be great for the calendar - You see you could have the year and than the folder per month...

Like written earlier it's just an idea as I miss that option for my usage but in global I love your work so far

**Reply by graciousgrey** [reply]
> Hi so far I'm happy and so I'm started with a sponsoring

:heart: 

> I swapped to German but the categories were created in English - I simply can rename them?

Labels are not yet translated automatically. It's planned for the future (https://github.com/photoprism/photoprism/issues/485). Until then you can simply rename them. It might take some minutes until titles get updated by the background worker.

For wrong labels it depends, if you have the feeling that all photos labeled as weapon are industrial buildings you can rename it. If it is just wrong for single photos removing is better (https://docs.photoprism.app/user-guide/organize/labels/#removedelete-labels-from-photos). Renaming will always be applied to all photos with this label.

> Even under folders it's not displayed like the folders are but just on one level...

We plan to add an option to display folders hierarchically --> https://github.com/photoprism/photoprism/issues/1536

> Same would be great for the calendar - You see you could have the year and than the folder per month...

For the calendar it is not planned. A fixed hierarchy would only make sense for year>month. Using the search is more flexible. You can search for "2020" to see all calendar-albums of 2020 or search for "may" to see all calendar-albums of may. Searching for german months e.g.  "mai" does not work yet but can be improved eventually.
